File: 01-programming-fundamentals/03-contest-01/cpp/katryoshka.cpp
#include<bits/stdc++.h>
using namespace std;

int main() {
    long long e, m, b;
    cin >> e >> m >> b;
    long long res=0;
    if(e!=0 || b!=0) {
        if(m<e && m<b) {
            res = m;
            e = e - m;
            b = b - m;
            res += min(e/2, b);
        } else {
            res = min(e, b);
        }
    }

    cout << res << endl;
    return 0;
}

File: 01-Programming-Basics/02-Contest-1/Katryoshka.java
import java.util.Scanner;

public class Katryoshka {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        long e = sc.nextLong();
        long m = sc.nextLong();
        long b = sc.nextLong();

        long res = 0;
        
        long thirdReceipe  = Math.min(e, Math.min(m, b));
        res += thirdReceipe;

        long firstReceipe = Math.min((e-thirdReceipe)/2, b-thirdReceipe);
        res += firstReceipe;

        System.out.println(res);
    }
}
